onadata/apps/logger/models/xform.py

```diff
@@ -616,15 +626,33 @@ def _check_version_set(self, survey):
         the default version in this datetime (yyyymmddhhmm) format.
         """
 
+        change_version = False
         # get the json and check for the version key
         survey_json = json.loads(survey.to_json())
-        if not survey_json.get("version"):
+        version = survey_json.get("version")
+        if version is None:
             # set utc time as the default version
-            survey_json['version'] = \
-                datetime.utcnow().strftime("%Y%m%d%H%M")
+            change_version = True
+            version = create_xform_version()
+        else:
+            try:
+                int(version)
+            except ValueError:
+                raise XLSFormError(
+                    _(u"version should be a string of upto 10 numbers that "
+                      "describe this revision. You could use a string of the "
+                      "form 'yyyymmddrr'. For example, 2017021501 is the 1st "
```

Comment:
Is `rr` a standard? I think some text describing that is in order since I don't expect to see that in what looks like a date format string.

Also, if we think this is the best I think we should be more forceful then "could", e.g. replacing the sentance beginning with "You could" to address both:

> Please use a string of the form 'yyyymmddrr', where `rr` is the revision number with a leading zero if less than 10.